lib/archiving.js

```diff
@@ -169,10 +169,11 @@ function Archive(config, properties) {
 
 api = function (config, method, path, body, callback) {
   var rurl = config.apiEndpoint + '/v2/project/' + config.apiKey + path;
+  var reqWithDefaults = request;
```

Comment:
@rviscarra How will this change the behavior? The `reqWithDefaults` variable will point to the same request instance that the `request` variable points to.

Reply:
@jeffswartz According to the request [docs.](https://github.com/request/request#requestdefaultsoptions), `request.defaults` returns a wrapped instance that uses the specified default values. If the variable is not declared locally the request object is wrapped around on every request, after enough API calls/wraps it starts failing with `Maximum call stack size exceeded`

Reply:
Thanks for the clarification, @rviscarra. If the wrapped instance returned by `request.defaults()` is a new instance, can we please eliminate the `reqWithDefaults` variable and do something like this:

```
api = function (config, method, path, body, callback) {
  var rurl = config.apiEndpoint + '/v2/project/' + config.apiKey + path;
  request.defaults(_.pick(config, 'proxy', 'timeout'))
    .request({
      url: rurl,
      method: method,
      headers: generateHeaders(config),
      json: body
    }, callback);
};
```

Note also that `'defaults' in request` will always be `true`, correct?

Reply:
Yes, it should always be `true` since the version is pinned down. I'll update the PR!

Reply:
@jeffswartz, I didn't notice you second `.request` reference there, it would have to be something like this:

```javascript
request.defaults(_.pick(config, 'proxy', 'timeout'))({
  url: rurl,
  method: method,
  headers: generateHeaders(config),
  json: body
}, callback);
```
since `defaults` returns a function (which is what `request` originally is).

Reply:
Yes, that's right. (My second `.request()` reference was incorrect.) Can you please update the PR?